src/charm.py

```diff
@@ -105,15 +104,19 @@ def _on_mimir_pebble_ready(self, event):
     def _on_config_changed(self, event):
         self._configure(event)
 
+    def _on_stop(self, _):
+        # Clear the workload version in case something's wrong after an upgrade.
+        self.unit.set_workload_version("")
+
     def _configure(self, event):
         if not self._container.can_connect():
             self.unit.status = WaitingStatus("Waiting for Pebble ready")
             return
 
         try:
+            self._set_alerts()
```

Comment:
Do we need to add an api call or the workload looks for changes?

Reply:
The workload doesn't look for changes unless it's local. When we stop using it, it requires an API call or mimirtool

Reply:
I'll put that in a comment.